### PR TITLE
refactor(CO-764): remove 2FA, Devices, Trusted devices, App Specific Password, Remote Wipe

### PR DESCRIPTION
**Pull request by frisonisland**
This PRs removes 2FA logic and Devices + Trusted Devices and App Specific Passwords since it is not provided anymore.
**Plan to merge it for 23.8-23.9**

2FA:
- auth logic
- scratch codes

App specific password
- auth logic

Devices:
- verification code
- CRUD devices
- reminder device
- devices count
- sync

Trusted devices:
- auth logic
- token

Remote Wipe was part of Network Edition, so it was removed.
**The PR is quite big, but mostly because of deleted files.**
**I ask to please carefully check:**
- [admin Auth.java](https://github.com/zextras/carbonio-mailbox/pull/258/files#diff-7a16be07dac9a47a1769524350df860de80e4c1ece289a8da31103cbda60331e)
- https://github.com/zextras/carbonio-mailbox/pull/258/files#diff-7a16be07dac9a47a1769524350df860de80e4c1ece289a8da31103cbda60331e
- https://github.com/zextras/carbonio-mailbox/pull/258/files#diff-d79855035c7ea2d1dfe6889ee4796f0badea1aa28eac05f28792801778c908d3
- https://github.com/zextras/carbonio-mailbox/pull/258/files#diff-9d49b27d5fa0d6a57634e1382de08ffaa7fd2f6aa02d56b6d2e44be0925fcbf8
- https://github.com/zextras/carbonio-mailbox/pull/258/files#diff-b06e5582dd5e9c7544d9e7057d463c9242fb64fe9d6c4c8f46a92922b165a25a
- name of deleted request/response classes

**Manual testing**:
- zmsoap (was giving _ERROR: service.AUTH_REQUIRED (no valid authtoken present)_) fixed in https://github.com/zextras/carbonio-mailbox/pull/258/commits/b57cd85cf8b16a56b780a3a8d054e0e0763fd7d9
- IMAP (Thunderbird)

**TODO:**
- [x] cleanup https://github.com/zextras/carbonio-db-conf/blob/main/src/db/mysql/db.sql#L211 from devices

### CODE REVIEW

File: store/utils/docker/carbonio-db/scripts/01-db.sql [hunk 1]
```diff
@@ -207,40 +207,6 @@ CREATE TABLE scheduled_task (
    INDEX i_mailbox_id (mailbox_id)
 ) ENGINE = InnoDB;
 
--- Mobile Devices
```
Comment (keshavbhatt) [inline store/utils/docker/carbonio-db/scripts/01-db.sql:210]: We should also perform this cleanup in actual schema too
https://github.com/zextras/carbonio-db-conf/blob/main/src/db/mysql/db.sql#L211

File: store/src/main/java/com/zimbra/cs/account/AuthToken.java [hunk 2]
```diff
@@ -256,7 +256,7 @@ public static Map getInfo(String encoded) throws AuthTokenException {
     public abstract Usage getUsage();
 
     public static enum Usage {
```
Comment (keshavbhatt) [inline store/src/main/java/com/zimbra/cs/account/AuthToken.java:258]: We can also remove static from this enum definition.

**Review by keshavbhatt — APPROVED** [review]
LGTM. 

I asked for one minor change and asked one question.

File: store/src/main/java/com/zimbra/cs/account/AuthToken.java (hunk 2 above)
Comment (frisonisland) [reply store/src/main/java/com/zimbra/cs/account/AuthToken.java]: Done!

**Review by Polpetta — APPROVED** [review]
LGTM

**Comment by Polpetta** [comment]
@frisonisland did you remove the Yahoo auth too here?

**Comment by frisonisland** [comment]
> @frisonisland did you remove the Yahoo auth too here?

Nope!

**Pull request merged**